tests/components/blebox/conftest.py

```diff
@@ -0,0 +1,148 @@
+"""PyTest fixtures and test helpers."""
+
+from unittest import mock
+
+from asynctest import CoroutineMock, call, patch
+import blebox_uniapi
+import pytest
+
+from homeassistant.components.blebox import const
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.exceptions import PlatformNotReady
+
+from tests.common import MockConfigEntry
+
+
+def setup_product_mock(
+    category, feature_mocks, path="homeassistant.components.blebox.Products"
+):
+    """Mock a product returning the given features."""
+
+    product_mock = mock.create_autospec(
+        blebox_uniapi.box.Box, True, True, features={category: feature_mocks}
+    )
+    patcher = patch(path, mock.DEFAULT, blebox_uniapi.products.Products, True, True)
+    products_class = patcher.start()
+    products_class.async_from_host = CoroutineMock(return_value=product_mock)
+    return product_mock
+
+
+def mock_only_feature(spec, **kwargs):
+    """Mock just the feature, without the product setup."""
+    return mock.create_autospec(spec, True, True, **kwargs)
+
+
+def mock_feature(category, spec, **kwargs):
+    """Mock a feature along with whole product setup."""
+    feature_mock = mock_only_feature(spec, **kwargs)
+    setup_product_mock(category, [feature_mock])
+    return feature_mock
+
+
+def mock_config(ip_address="172.100.123.4"):
+    """Return a Mock of the HA entity config."""
+    return MockConfigEntry(
+        domain=const.DOMAIN, data={CONF_HOST: ip_address, CONF_PORT: 80},
+    )
+
+
+class DefaultBoxTest:
```

Comment:
New tests need to be standalone pytest test functions (no classes).

Reply:
As mentioned elsewhere, I need this class for shared tests (used in other platforms, removed from this PR) and I can't figure out how properly implement shared tests.